de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -161,6 +168,77 @@ protected SWTBotTreeItem openEditor(SWTBotTreeItem item) {
 		return newItem;
 	}
 	
+	/**
+	 * @param item Tree item for which a diagram editor gets opened
+	 * @return Graphiti GEF Editor handle that SWTBot can work on
+	 */
+	protected SWTBotGefEditor openDiagramEditorForTreeItem(SWTBotTreeItem item) {
+		SWTGefBot gefBot = new SWTGefBot();
+		item.contextMenu().menu("Open Diagram Editor").click();
+		waitForEditingDomainAndUiThread();		
+		String editorTitle = bot.editors().get(1).getTitle();				
+		SWTBotGefEditor editor = gefBot.gefEditor(editorTitle);
+		return editor;
+	}
+	
+	/**

```

Comment:
We need some comments here on this method and how it works since it does some crazy reflection stuff.